internal/options.ily

```diff
@@ -105,45 +105,66 @@
 
 % First the necessary predicates
 #(define (prop-rule? obj)
-   "Check if obj is a property rule. Property rules must satisfy:
-    - be a list
-    - first element is a symbol? (the key)
-    - second element - if present -  must be a procedure? (a predicate)
-   (- third element would be the default, but isn't checked in this predicate)"
-   (and (list? obj)
-        (symbol? (first obj))
-        (or (= (length obj) 1)
-            (eq? 'opt (last obj))
-            (procedure? (second obj)))))
+   "Check if obj is a property rule. A property rule can have 5 forms:
+    - arg-name                    (a symbol, stating that this argument is required)
+    - (arg-name)                  (a list with a symbol, same as above)
+    - (arg-name ,type?)           (same as above with a type check)
+    - (? arg-name)                (optional argument, for strict rulesets)
+    - (? arg-name ,type?)         (same as above with a type-check)
+    - (? arg-name ,type? def-v)   (same as above with a default value)
+    Default values aren't checked in this predicate"
+   (or (symbol? obj)
+       (and (list? obj)
+            (let ((l (length obj))
+                  (f (first obj)))
+              (and (symbol? f)
+                   (or (= l 1)
+                       (and (= l 2)
+                            (if (eq? '? f)
+                                (symbol? (second obj))
+                                (procedure? (second obj))))
+                       (and (> l 2)
+                            (< l 5)
+                            (symbol? (second obj))
```

Comment:
Shouldn't this also have the `(if (eq? '? f)` conditional like in the previous check?

Reply:
Now that you mention it, yes, and some kind of check for `'?` should also appear for the `(= l 1)` and the `(symbol? obj)` cases.